plugins/colorbutton/plugin.js

```diff
@@ -203,6 +203,11 @@ CKEDITOR.plugins.add( 'colorbutton', {
 				total = colors.length + ( moreColorsEnabled ? 2 : 1 );
 
 			var clickFn = CKEDITOR.tools.addFunction( function applyColorStyle( color, type ) {
+				// #565
+				if ( /^[0-9a-f]{3}([0-9a-f]{3})?$/i.test( color ) ) {
+					color = '#' + color;
```

Comment:
IMO the fix is placed in incorrect plugin. I'd move it into `colordialog` one, inside `onClose` handler of the dialog.